src/fu-engine.c

```diff
@@ -1054,6 +1054,110 @@ fu_engine_check_hardware_requirement (FuEngine *self, AsApp *app, GError **error
 }
 #endif
 
+#if AS_CHECK_VERSION(0,7,8)
+static gchar *
+fu_engine_get_pkgconfig_version_fn (const gchar *pkgconfig_fn, GError **error)
+{
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) lines = NULL;
+
+	/* load file */
+	if (!g_file_get_contents (pkgconfig_fn, &data, NULL, error))
+		return NULL;
+
+	/* look for version */
+	lines = g_strsplit (data, "\n", -1);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		if (g_str_has_prefix (lines[i], "Version: "))
+			return g_strdup (lines[i] + 9);
+	}
+
+	/* nothing suitable found! */
+	g_set_error (error,
+		     FWUPD_ERROR,
+		     FWUPD_ERROR_INVALID_FILE,
+		     "Invalid pkg-config file found: %s",
+		     pkgconfig_fn);
+	return FALSE;
+}
+
+static gchar *
+fu_engine_get_pkgconfig_version (const gchar *pkgconfig_name, GError **error)
+{
+	g_autofree gchar *fn = g_strdup_printf ("%s.pc", pkgconfig_name);
+	const gchar *paths[] = { LIBDIR, DATADIR, "/usr/share", "/usr/lib64", NULL };
```

Comment:
Missing a few paths (the local ones and ones that might have been set by env variable).  (from man page):
```
       pkg-config retrieves information about packages from special metadata files. These files are named after the package, and  has  a  .pc  extension.   On  most  systems,  pkg-config  looks  in  /usr/lib/pkgconfig,
       /usr/share/pkgconfig, /usr/local/lib/pkgconfig and /usr/local/share/pkgconfig for these files.  It will additionally look in the colon-separated (on Windows, semicolon-separated) list of directories specified by
       the PKG_CONFIG_PATH environment variable.
```